node/pkg/txverifier/evm.go

```diff
@@ -181,12 +181,101 @@ func (tv *TransferVerifier[ethClient, Connector]) pruneCache() {
 	}
 
 	if numPrunedEvals > 0 {
-		tv.logger.Debug("pruned cached receipt evaluations",
+		tv.logger.Info("pruned receipt evaluations cache",
 			zap.Int("numPruned", numPrunedEvals),
 			zap.Int("cacheSize", len(tv.evaluations)),
 		)
 	}
-	// TODO: The other caches should be pruned here too.
+
+	// For the other caches, just prune them if they're too big.
+	if len(tv.chainIdCache) >= CacheMaxSize {
+		var (
+			i = 0
+			// Slice to store keys to delete in order to avoid deleting keys from maps that are being
+			// iterated over.
+			keysToDelete = make([]string, CacheDeleteCount)
+		)
+		for key := range tv.chainIdCache {
```

Comment:
This is where the LRU cache would be good to avoid a scenario where the first X keys are the most common and get pruned every time. Nice suggestion to add that in a follow up PR!